### PR TITLE
Adding DPUB use cases

### LINKED ISSUE #1670: Should "Link has non-empty name" apply to inheriting links? [c487ae]
I noticed this rule applies to semantic links. Seems to me this should also apply to inheriting semantic links.

On the other hand though, should we expect anyone testing HTML to support DPUB? Should that be in a separate rule?

### PR DESCRIPTION
**Pull request by daniel-montalvo**
Adds DPUB use cases.

Closes issue(s):

- closes #1670 

Need for Call for Review:
This will require a 1 week Call for Review << small changes affecting a small number of test cases, if in doubt do not use this. >>

---

### **When creating PR:**

- [x] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [x] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [x] Add yourself (and co-authors) as "Assignees" for PR.
- [x] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [x] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [ ] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 1]
```diff
@@ -185,6 +185,14 @@ This `area` element has a [semantic role][] of `link` and an [accessible name][]
 </map>
 ```
 
+#### Passed Example 11
+
+This `span` element has an [inheriting semantic role] of `link` and an [accessible name] given by its child text node.
+
+```html
+See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref link">ACT rules</span
```
Comment (daniel-montalvo) [inline _rules/link-non-empty-accessible-name-c487ae.md:193]:
```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>]
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 2]
```diff
@@ -274,6 +282,14 @@ This `a` element has an [explicit role][] of `none`. However, it is [focusable][
 <a href="https://www.w3.org/WAI" role="none"> </a>
 ```
 
+### Failed Example 11
+
+This `span` element has an [inheriting semantic role] of `link` and an empty [accessible name].
+
+```html
+See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref link"></span
```
Comment (daniel-montalvo) [inline _rules/link-non-empty-accessible-name-c487ae.md:290]:
```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></span>]
```

**Review by colabottles — APPROVED** [review]

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 3]
```diff
@@ -185,6 +185,14 @@ This `area` element has a [semantic role][] of `link` and an [accessible name][]
 </map>
 ```
 
+#### Passed Example 11
+
+This `span` element has an [inheriting semantic role][] of `link` and an [accessible name] given by its child text node.
+
+```html
+See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>]
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:193]: I'm a little hesitant to say this passes SC 4.1.2. This wouldn't be accessibility supported. Perhaps what we need to do is put these on actual anchor elements?

**Review by WilcoFiers — DISMISSED** [review]
Couple of points:
- You'll need to update the applicability, so that it uses inheriting semantic role.
- You should probably also update the acc support section to point out that these doc-* roles aren't widely supported.
- Do you want to add yourself to the authors list?

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 4]
```diff
@@ -185,6 +186,14 @@ This `area` element has a [semantic role][] of `link` and an [accessible name][]
 </map>
 ```
 
+#### Passed Example 11
+
+This `span` element has an [inheriting semantic role][] of `link` and an [accessible name] given by its child text node.
+
+```html
+See [<a onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</a>]
```
Comment (Jym77) [inline _rules/link-non-empty-accessible-name-c487ae.md:194]: This is not a `span` element 🙈 
```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>].
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 5]
```diff
@@ -274,6 +283,14 @@ This `a` element has an [explicit role][] of `none`. However, it is [focusable][
 <a href="https://www.w3.org/WAI" role="none"> </a>
 ```
 
+#### Failed Example 11
+
+This `span` element has an [inheriting semantic role][] of `link` and an empty [accessible name].
+
+```html
+See [<a onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></a>]
```
Comment (Jym77) [inline _rules/link-non-empty-accessible-name-c487ae.md:291]:
```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></span>].
```

Comment (Jym77) [reply _rules/link-non-empty-accessible-name-c487ae.md:291]: Maybe adding an image inside the `span` here. Visually, nothing is displayed and there is almost nowhere to actually click, which is at least weird.
Adding [this image](https://github.com/act-rules/act-rules.github.io/blob/develop/test-assets/shared/act-logo.png) with an empty `alt` would be better, I think.

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 6]
```diff
@@ -332,6 +349,7 @@ This `a` element does not have the role of link because it does not have an `hre
 [explicit role]: #explicit-role 'Definition of Explicit Role'
 [focusable]: #focusable 'Definition of focusable'
 [included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
+[inheriting semantic role]: #inheriting-semantic-role 'Definition of inheriting semantic role'
```
Comment (Jym77) [inline _rules/link-non-empty-accessible-name-c487ae.md:352]:
```suggestion
[inheriting semantic role]: #inheriting-semantic 'Definition of inheriting semantic role'
```
The key is only `inheriting-semantic`, that should fix the test 🤞

**Review by Jym77 — DISMISSED** [review]

**Review by colabottles — APPROVED** [review]

**Comment by daniel-montalvo** [comment]
Hey @Jym77 

`span` elements were turned into `a` elements based on @wilcofiers  suggestion that `span` elements would probably not pass 4.1.2 in this context. I should change the description rather than the examples themselves. Would you not agree with this?

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 7]
```diff
@@ -45,7 +45,7 @@ htmlHintIgnore:
 
 ## Applicability
 
-This rule applies to any [HTML element][] that is a [semantic][semantic role] `link` [included in the accessibility tree][].
+This rule applies to any [HTML element][] that is an [inheriting semantic][inheriting semantic role] `link` [included in the accessibility tree][].
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:48]: Suggest you do them the other way around:

```suggestion
This rule applies to any [HTML element][] that is an [inheriting semantic][] `link` [included in the accessibility tree][].
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 8]
```diff
@@ -60,6 +60,7 @@ The rule assumes that all links are [user interface components](https://www.w3.o
 - There are assistive technologies that do not support using the `title` attribute for an [accessible name][], or in which this feature can be disabled.
 - For `area` elements that have an `href` attribute, but are not nested inside a `map` element, there are differences between browsers and assistive technology on if the `area` is [included in the accessibility tree][].
 - Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some [semantic][semantic role] `link` elements can fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+- Accessibility support for [inheriting semantic][inheriting-semantic] links may vary depending on the assistive technology in use.
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:63]:
```suggestion
- Accessibility support for [inheriting semantic][] links may vary depending on the assistive technology in use.
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 9]
```diff
@@ -185,6 +186,14 @@ This `area` element has a [semantic role][] of `link` and an [accessible name][]
 </map>
 ```
 
+#### Passed Example 11
+
+This `a` element has an [inheriting semantic role][] of `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:191]:
```suggestion
This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 10]
```diff
@@ -274,6 +283,14 @@ This `a` element has an [explicit role][] of `none`. However, it is [focusable][
 <a href="https://www.w3.org/WAI" role="none"> </a>
 ```
 
+#### Failed Example 11
+
+This `a` element has an [inheriting semantic role][] of `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:288]:
```suggestion
This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
Tests are failing. Suggest doing things like this. It keeps things consistent with rules like Link is descriptive.

**Review by colabottles — APPROVED** [review]

**Review by Jym77 — APPROVED** [review]

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 11]
```diff
@@ -45,7 +45,7 @@ htmlHintIgnore:
 
 ## Applicability
 
-This rule applies to any [HTML element][] that is a [semantic][semantic role] `link` [included in the accessibility tree][].
+This rule applies to any [HTML element][] that is an [inheriting semantic][inheriting semantic] `link` [included in the accessibility tree][].
```
Comment (WilcoFiers) [inline _rules/link-non-empty-accessible-name-c487ae.md:48]: Think you can simplify this:

```suggestion
This rule applies to any [HTML element][] that is an [inheriting semantic][] `link` [included in the accessibility tree][].
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 12]
```diff
@@ -60,6 +60,7 @@ The rule assumes that all links are [user interface components](https://www.w3.o
 - There are assistive technologies that do not support using the `title` attribute for an [accessible name][], or in which this feature can be disabled.
 - For `area` elements that have an `href` attribute, but are not nested inside a `map` element, there are differences between browsers and assistive technology on if the `area` is [included in the accessibility tree][].
 - Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some [semantic][semantic role] `link` elements can fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+- Accessibility support for [inheriting semantic][] links may vary depending on the assistive technology in use.
```
Comment (carlosapaduarte) [inline _rules/link-non-empty-accessible-name-c487ae.md:63]:
```suggestion
- Accessibility support for some elements inheriting the semantic role of `link` (e.g. elements with `doc-*` attributes) may vary depending on the assistive technology in use.
```

**Review by carlosapaduarte — COMMENTED** [review]
I understand why you are not convinced by the accessibility support note you added. I've tried to update it to better match Wilco's concerns. I'll leave it to your judgment on whether this is useful or not.

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 13]
```diff
@@ -185,6 +186,14 @@ This `area` element has a [semantic role][] of `link` and an [accessible name][]
 </map>
 ```
 
+#### Passed Example 11
+
+This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```
Comment (carlosapaduarte) [inline _rules/link-non-empty-accessible-name-c487ae.md:191]: To be consistent with the way the applicability is written. 

```suggestion
This `a` element is an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```

File: _rules/link-non-empty-accessible-name-c487ae.md [hunk 14]
```diff
@@ -274,6 +283,16 @@ This `a` element has an [explicit role][] of `none`. However, it is [focusable][
 <a href="https://www.w3.org/WAI" role="none"> </a>
 ```
 
+#### Failed Example 11
+
+This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```
Comment (carlosapaduarte) [inline _rules/link-non-empty-accessible-name-c487ae.md:288]: To be consistent with the way the applicability is written. 

```suggestion
This `a` element is an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```

**Review by carlosapaduarte — CHANGES_REQUESTED** [review]
editorial

**Review by colabottles — APPROVED** [review]

File: _rules/link-non-empty-accessible-name-c487ae.md (hunk 12 above)
Comment (daniel-montalvo) [inline _rules/link-non-empty-accessible-name-c487ae.md:63]:
```suggestion
- Accessibility support for some elements inheriting the semantic role of `link` (e.g. elements with `doc-*` attributes) may vary depending on the assistive technology in use.
```
I seem to have dropped this accidentally @carlosapaduarte  Bringing it back.

**Review by WilcoFiers — APPROVED** [review]

**Review by kherr-a11y — APPROVED** [review]

**Pull request merged**